src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -1952,6 +1952,48 @@ void parser::parse_and_visit_function_parameters(parse_visitor_base &v,
         parameter_property_keyword = readonly_span;
         this->lexer_.commit_transaction(std::move(transaction));
       }
+
+      // consturctor(readonly public value) {}  // Invalid
+      // constructor(public readonly value) {}  // Ok, TypeScript only
+      // constructor(readonly public) {}        // Ok, TypeScript only
+                                                // public used as identifier
+      if (this->options_.typescript) {
+        switch (this->peek().type) {
+          case token_type::kw_private:
+          case token_type::kw_public:
+          case token_type::kw_protected: {
+            source_code_span specifier_span = this->peek().span();
+            parser_transaction transaction = this->begin_transaction();
+            this->skip();
+
+            switch (this->peek().type) {
+              QLJS_CASE_STRICT_ONLY_RESERVED_KEYWORD:
+              QLJS_CASE_CONTEXTUAL_KEYWORD:
+              case token_type::kw_await:
+              case token_type::dot_dot_dot:
+              case token_type::identifier:
+              case token_type::kw_yield:
+              case token_type::left_curly:
+              case token_type::left_square:
+              case token_type::reserved_keyword_with_escape_sequence: {
```

Comment:
Is this copy-pasted from somewhere? I don't see comprehensive tests justifying each of these cases.

If it's copy-pasted, could we avoid the copy-paste somehow? If we can't reasonbly avoid the copy-paste, I'd like a test for each of these cases.

Reply:
Yeah, that was a dirty copy paste which I wanted to refactor and I've found a simpler way to check if access modifier after "readonly" is used as a keyword or identifier by simply skipping a token and checking with already existing lambda function 'is_after_parameter_name' if access modifier was used as an identifier.